File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/infrastructure/llm/PromptBuilder.java
package com.gonezo.multiplatform.audioextraction.infrastructure.llm;

import com.gonezo.multiplatform.audioextraction.domain.model.ExecutionPlan;
import com.gonezo.multiplatform.audioextraction.domain.model.Transcript;
import com.gonezo.multiplatform.audioextraction.domain.schema.OutputSchema;

public interface PromptBuilder {
  String build(Transcript transcript, OutputSchema schema, ExecutionPlan plan);
}

File: core/bin/main/com/gonezo/audioextraction/infrastructure/llm/PromptBuilder.kt
package com.gonezo.audioextraction.infrastructure.llm

import com.gonezo.audioextraction.domain.model.ExecutionPlan
import com.gonezo.audioextraction.domain.model.Transcript
import com.gonezo.audioextraction.domain.schema.OutputSchema

interface PromptBuilder {
  fun build(transcript: Transcript, schema: OutputSchema, plan: ExecutionPlan): String
}
